example.go

```diff
@@ -39,46 +42,51 @@ var (
 
 var (
 	subscriptionID string
-	spToken        *azure.ServicePrincipalToken
+	authorizer     *autorest.BearerAuthorizer
 )
 
 func init() {
 	subscriptionID = getEnvVarOrExit("AZURE_SUBSCRIPTION_ID")
 	tenantID := getEnvVarOrExit("AZURE_TENANT_ID")
 
-	oauthConfig, err := azure.PublicCloud.OAuthConfigForTenant(tenantID)
+	oauthConfig, err := adal.NewOAuthConfig(azure.PublicCloud.ActiveDirectoryEndpoint, tenantID)
 	onErrorFail(err, "OAuthConfigForTenant failed")
 
 	clientID := getEnvVarOrExit("AZURE_CLIENT_ID")
 	clientSecret := getEnvVarOrExit("AZURE_CLIENT_SECRET")
-	spToken, err = azure.NewServicePrincipalToken(*oauthConfig, clientID, clientSecret, azure.PublicCloud.ResourceManagerEndpoint)
+	spToken, err := adal.NewServicePrincipalToken(*oauthConfig, clientID, clientSecret, azure.PublicCloud.ResourceManagerEndpoint)
+	authorizer = autorest.NewBearerAuthorizer(spToken)
 	onErrorFail(err, "NewServicePrincipalToken failed")
 
 	createClients()
 }
 
 func main() {
 	fmt.Println("Creating resource group")
-	resourceGroupParameters := resources.ResourceGroup{
+	resourceGroupParameters := resources.Group{
 		Location: &westus}
 	_, err := groupClient.CreateOrUpdate(groupName, resourceGroupParameters)
 	onErrorFail(err, "CreateOrUpdate failed")
 
-	fmt.Println("Creating storage account")
-	// apcp := storage.AccountPropertiesCreateParameters{}
+	fmt.Println("Starting to create storage account...")
 	accountParameters := storage.AccountCreateParameters{
 		Sku: &storage.Sku{
 			Name: storage.StandardLRS,
 		},
 		Kind:     storage.Storage,
 		Location: &westus,
 		AccountPropertiesCreateParameters: &storage.AccountPropertiesCreateParameters{},
-		// AccountPropertiesCreateParameters: &apcp,
 	}
-	_, err = accountClient.Create(groupName, storageAccountName, accountParameters, nil)
-	onErrorFail(err, "Create failed")
-
-	fmt.Println("Creating public IP address")
+	_, errStorageAccount := accountClient.Create(groupName, storageAccountName, accountParameters, nil)
+	var wg sync.WaitGroup
```

Comment:
Per our conversation I think this can be removed and just read from the error channel.